lib/gis/strings.c

```diff
@@ -250,6 +251,45 @@ char *G_str_replace(const char *buffer, const char *old_str, const char *new_str
     return replace;
 }
 
+/*!
+  \brief String concatenation
+
+  Concatenates the strings in src_strings, which consists of num_strings number
+  of strings, with the separator sep. The size of the concatenated string is
+  limited by maxsize.
+
+  \param src_strings array of strings to concatenate
+  \param num_strings count of strings in src_strings
+  \param sep separator string
+  \param maxsize maximum number of characters of returned string
+
+  \return the concatenated string (allocated)
+*/
+char *G_str_concat(const char **src_strings, int num_strings,
+		   const char *sep, int maxsize)
+{
+    char buffer[maxsize];
+    int i;
+    char* end = buffer + maxsize;
+    char* p = NULL;
+
+    if (maxsize < 1 || num_strings < 1)
+	return NULL;
+
+    memset(buffer, 0, sizeof(buffer));
+
+    for (i = 0; i < num_strings; i++) {
+	if (i == 0)
```

Comment:
Please check our indentation guidelines at https://trac.osgeo.org/grass/wiki/Submitting/C#Indentation. Here, tabs with four and two spaces are mixed.